synapse/lib/nexus.py

```diff
@@ -263,145 +263,75 @@ async def fini():
 
             yield dist
 
-    async def isLeader(self):
-        return self._ldrurl is None
-
-    async def setLeader(self, url: Optional[str], iden: str) -> None:
-        '''
-        Args:
-            url:  if None, sets this nexsroot as leader, otherwise the telepath URL of the leader (must be a Cell)
-            iden: iden of the leader.  Should be the same as my containing cell's iden
-        '''
-        if url is not None and not self.donexslog:
-            raise s_exc.BadConfValu(mesg='Mirroring incompatible without nexslog:en')
-
-        former = self._ldrurl
-
-        if former == url:
-            return
-
-        self._ldrurl = url
-
-        oldlooptask = self._looptask
-
-        # If the looptask is already running, stop it
-        if self._looptask is not None:
-            self._looptask.cancel()
-            self._looptask = None
-            self._ldrready.clear()
-            if self._ldr is not None:
-                await self._ldr.fini()
-            self._ldr = None
+    async def startup(self, mirurl):
 
-        await self._dopreleader()
-        await self._dostatechange()
+        if self.client is not None:
+            await self.client.fini()
 
-        if not oldlooptask:
-            # Before we start mirroring anything, replay the last event because we don't know if it got committed
-            await self.recover()
+        self.client = None
+        if mirurl is not None:
+            self.client = await s_telepath.Client.anit(mirurl, onlink=self._onTeleLink)
+            self.onfini(self.client)
 
-        if self._ldrurl is None:
-            return
+        self.started = True
 
-        self._looptask = self.schedCoro(self._followerLoop(iden))
+    async def promote(self):
```

Comment:
docstring. why would i call this?

Reply:
This moves us toward the ability to runtime-promote a mirror into being the primary to minimize downtime when failing-over to a mirror.